roaring/roaring_test.go

```diff
@@ -1137,3 +1112,113 @@ func TestBitmap_Intersect(t *testing.T) {
 		t.Fatalf("Counts do not match %d %d", bm0.Count(), result.Count())
 	}
 }
+
+func BenchmarkBitmap_IntersectionCount_ArrayBitmap(b *testing.B) {
+	data := &benchmarkBitmapIntersectionCountData
+	if data.a == nil {
+		const max = (1 << 24) / 64
+
+		// Build bitmap with array container.
+		data.a = roaring.NewBitmap()
+		for i, n := 0, rand.Intn(roaring.ArrayMaxSize); i < n; i++ {
+			data.a.Add(uint64(rand.Intn(max)))
+		}
+
+		// Build bitmap with bitmap container.
+		data.b = roaring.NewBitmap()
+		for i, n := 0, roaring.ArrayMaxSize*2; i < n; i++ {
+			data.b.Add(uint64(i * 3))
+		}
+	}
+
+	// Reset timer & benchmark.
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		data.a.IntersectionCount(data.b)
+	}
+}
+
+const (
+	NumRows         = uint64(10000)
+	NumColums       = uint64(4)
+	MaxContainerVal = 0xffff
+)
+
+func BenchmarkContainerLinear(b *testing.B) {
+
+	for n := 0; n < b.N; n++ {
+		bm := roaring.NewBitmap()
+		for row := uint64(0); row < NumRows; row++ {
+			for col := uint64(0); col < NumColums; col++ {
+				bm.Add(row*pilosa.SliceWidth + (col * MaxContainerVal))
+			}
+		}
+	}
+}
+
+func BenchmarkContainerReverse(b *testing.B) {
+	for n := 0; n < b.N; n++ {
+		bm := roaring.NewBitmap()
+		for row := NumRows; row > 0; row-- {
```

Comment:
should probably do NumRows-1 and `row >= 0` for consistency with the bits set in the linear case. Same with the columns line below.